__tests__/app.test.js

```diff
@@ -0,0 +1,42 @@
+const app = require("../app");
+const request = require("supertest");
+const seed = require("../db/seeds/seed");
+const db = require("../db/connection");
+const data = require("../db/data/test-data/index");
+
+beforeEach(() => {
+  return seed(data);
+});
+
+afterAll(() => {
+  return db.end();
+});
+
+describe("GET /api/categories", () => {
+    test("GET - status:200, responds with an array of categories objects", () => {
+      return request(app)
+        .get("/api/categories")
+        .expect(200)
+        .then(({ body }) => {
+          const { categories } = body;
+          expect(categories).toBeInstanceOf(Array);
+          expect(categories).toHaveLength(4);

```

Comment:
Technically we don't need line 22 as we are using a forEach on line 24 (forEach only works on arrays and therefore implies the datatype), but it's not really an issue